kola/tests/misc/selinux.go

```diff
@@ -101,42 +143,62 @@ func SelinuxEnforce(c cluster.TestCluster) {
 
 // SelinuxBoolean checks that you can tweak a boolean in the current session
 func SelinuxBoolean(c cluster.TestCluster) {
-	cmdList := []cmdCheckOutput{
-		{"getsebool virt_use_nfs", true, ".*off"},
-		{"sudo setsebool virt_use_nfs 1", false, ""},
-		{"getsebool virt_use_nfs", true, ".*on"},
-	}
+	var seBoolean string = "virt_use_nfs"
 
 	m := c.Machines()[0]
 
+	newBool, err := toggleSelinuxBoolean(c, m, seBoolean)
+	if err != nil {
+		c.Fatalf(`Failed to generate toggled SELinux boolean: %v`, err)
+	}
+
+	// construct a regexp that looks like ".*off" or ".*on"
+	newBoolRegexp := ".*" + newBool[2]
+	cmdList := []cmdCheckOutput{
+		{"sudo setsebool " + seBoolean + " " + newBool[1], false, ""},
+		{"getsebool " + seBoolean, true, newBoolRegexp},
+	}
+
 	testSelinuxCmds(c, m, cmdList)
 
 	// since we didn't persist the change, should return to default value
-	output := c.MustSSH(m, "getsebool virt_use_nfs")
+	postOut := c.MustSSH(m, "getsebool "+seBoolean)
+	postBool := strings.Split(string(postOut), " ")[2]
 
-	if string(output) != "virt_use_nfs --> off" {
-		c.Fatalf(`The SELinux boolean "virt_use_nfs" is incorrectly configured: want %q, got %q`, "virt_use_nfs --> off", string(output))
+	// newBool[0] contains the original value of the boolean
+	if postBool != newBool[0] {
+		c.Fatalf(`The SELinux boolean "%q" is incorrectly configured: wanted %q, got %q`, seBoolean, newBool[0], postBool)
 	}
 }
 
 // SelinuxBooleanPersist checks that you can tweak a boolean and have it
 // persist across reboots
 func SelinuxBooleanPersist(c cluster.TestCluster) {
-	cmdList := []cmdCheckOutput{
-		{"getsebool virt_use_nfs", true, ".*off"},
-		{"sudo setsebool -P virt_use_nfs 1", false, ""},
-		{"getsebool virt_use_nfs", true, ".*on"},
-	}
+	var seBoolean string = "virt_use_nfs"
 
 	m := c.Machines()[0]
 
+	newBool, err := toggleSelinuxBoolean(c, m, seBoolean)
```

Comment:
This test looks almost identical to `SelinuxBoolean` ... but so was the previous version. Eventually it would be nice to reuse but for now :+1: